src/main/scala/scala/collection/decorators/BitSetDecorator.scala

```diff
@@ -44,8 +44,13 @@ class BitSetDecorator[+C <: BitSet with BitSetOps[C]](protected val bs: C) {
     val bitOffset = shiftBy & WordMask
     val wordOffset = shiftBy >>> LogWL
 
+    var significantWordCount = bs.nwords
+    while (significantWordCount > 0 && bs.word(significantWordCount - 1) == 0) {
+      significantWordCount -= 1
+    }
```

Comment:
Would it be possible to compute the amount to subtract from `significantWordCount` in `O(1)` time?

Reply:
I really doubt this. AFAIK `BitSet` implementation does not keep track of its most significant word/bit. Or maybe you have some idea?

Reply:
No, I haven’t thought about how to do this. This was just a question that popped up as I was quickly reading the diff.